docs/source/developers/java/development.rst

```diff
@@ -0,0 +1,170 @@
+    [INFO] Arrow AVRO Adapter ................................. SUCCESS [ 17.145 s]
+    [INFO] Arrow Algorithms ................................... SUCCESS [ 44.571 s]
+    [INFO] Arrow Performance Benchmarks ....................... SUCCESS [  9.202 s]
+    [INFO] Arrow Java C Data Interface ........................ SUCCESS [ 15.794 s]
+    [INFO] Arrow Orc Adapter .................................. SUCCESS [ 14.205 s]
+    [INFO] Arrow Gandiva ...................................... SUCCESS [02:06 min]
+    [INFO] Arrow Java Dataset ................................. SUCCESS [ 15.261 s]
+    [INFO] ------------------------------------------------------------------------
+    [INFO] BUILD SUCCESS
+    [INFO] ------------------------------------------------------------------------
+
+Performance Testing
+===================
+
+Java module that contains performance tests is: `performance`
+
+Let's configure our performance environment to run performance tests:
+
+- Install `benchmark`_
+- Install `archery`_
+- Install `conbench`_
+
+Lets execute benchmark tests:
+
+.. code-block::
+
+    $ cd benchmarks
+    $ conbench java-micro --help
+    $ conbench java-micro
+        --iterations=1
+        --commit=e90472e35b40f58b17d408438bb8de1641bfe6ef
+        --java-home=<absolute path to your java home>
+        --src=<absolute path to your arrow project>
+        --benchmark-filter=org.apache.arrow.adapter.AvroAdapterBenchmarks.testAvroToArrow
+    Benchmark                              Mode  Cnt       Score   Error  Units
+    AvroAdapterBenchmarks.testAvroToArrow  avgt       725545.783          ns/op
+    Time to POST http://localhost:5000/api/login/ 0.14911699295043945
+    Time to POST http://localhost:5000/api/benchmarks/ 0.06116318702697754
+
+Then go to: http://127.0.0.1:5000/ to see reports:
+
+UI Home:
+
+.. image:: img/conbench_ui.png
+
+UI Runs:
+
+.. image:: img/conbench_runs.png
+
+UI Benchmark:
+
+.. image:: img/conbench_benchmark.png
+
+Code Style
+==========
+
+Code style are configured at `java/dev/checkstyle`
+
+.. code-block::
+
+    <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-checkstyle-plugin</artifactId>
+        <version>3.1.0</version>
+        <executions>
+            <execution>
+                <id>validate</id>
+                <phase>validate</phase>
+                <goals>
+                    <goal>check</goal>
+                </goals>
+            </execution>
+        </executions>
+        <configuration>
+            <configLocation>dev/checkstyle/checkstyle.xml</configLocation>
+            <headerLocation>dev/checkstyle/checkstyle.license</headerLocation>
+            <suppressionsLocation>dev/checkstyle/suppressions.xml</suppressionsLocation>
+            <includeTestSourceDirectory>true</includeTestSourceDirectory>
+            ...
+        </configuration>
+    </plugin>
+
+Validate check style configuration:
```

Comment:
"The Maven build will validate code against the Checkstyle configuration. You can also just check the style without building the project:"

Maybe mention how to skip this check as well.

Reply:
Changed